src/main/java/com/ruralsourcing/WeatherBot/service/SlackIntegration.java

```diff
@@ -0,0 +1,99 @@
+package com.ruralsourcing.WeatherBot.service;
+
+import com.github.seratch.jslack.Slack;
+import com.github.seratch.jslack.api.model.event.MessageEvent;
+import com.github.seratch.jslack.api.rtm.RTMClient;
+import com.github.seratch.jslack.api.rtm.message.Message;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.ruralsourcing.WeatherBot.model.ResponseModel;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+
+/**
+ *
+ */
+public class SlackIntegration implements Runnable {
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    private static final String TOKEN = System.getenv("SLACK_BOT_API_TOKEN");
+
+    private final AtomicReference<ResponseModel> reference;
+
+    public SlackIntegration(AtomicReference<ResponseModel> reference) {
+        this.reference = reference;
+    }
+
+    @Override
+    public void run() {
+        rtmMessage();
+    }
+
+    private void rtmMessage() {
+        final Slack slack = Slack.getInstance();
+
+        try (final RTMClient rtm = slack.rtm(TOKEN)) {
+            rtm.addMessageHandler((message) -> {
+                final ResponseModel currentData = reference.get();
+
+                JsonObject json = JsonParser.parseString(message).getAsJsonObject();
+
+                if (json.get("type") == null) {
+                    return;
+                }
+
+                final String type = json.get("type").getAsString();
+
+                if(!"message".equals(type)){
+                    return;
+                }
+
+                final MessageEvent event = gson.fromJson(json, MessageEvent.class);
+
+                //do stuff in response:
+                System.out.println(event);
+
+                //If the message came from a DM or WeatherBot was pinged using @, respond
+                if(event.getText().contains("@UPXMYNW93") || event.getChannel().startsWith("D")){
```

Comment:
You can use the Slack API to retrieve your own user id rather than hard coding it. In fact, the `ConnectEvent` returned from the `rtm.connect` API should already return this data. Slack docs on that are [here](https://api.slack.com/methods/rtm.connect). and an example from heykudos is [here](https://github.com/DemonWav/heykudos/blob/master/main.go#L49-L51) and [here](https://github.com/DemonWav/heykudos/blob/4f6049d2bcf6ed32edf5a8c22e5438630d9f6c21/kudos.go#L32-L43).

Without knowing exactly how the library you're using works, I would expect that the first message you'd receive would be a `ConnectEvent` and would contain data relating to `self` (aka who the bot is).